weave

```diff
@@ -244,10 +264,16 @@ elif [ "$1" = "run" -a -z "$IS_LOCAL" ] ; then
         DNS_ARGS=$(exec_remote dns-args "$@" || true)
     fi
     shift $(dns_arg_count "$@")
+    if [ "$1" = "--no-rewrite-hosts" ] ; then
```

Comment:
We are now fixing a particular ordering of `--with-dns` and `--no-rewrite-hosts` at the beginning of the argument list. In other cases we took care to allow arguments in any order.